crates/util/src/util.rs

```diff
@@ -288,27 +290,21 @@ fn load_shell_from_passwd() -> Result<()> {
     Ok(())
 }
 
-#[cfg(unix)]
 /// Returns a shell escaped path for the current zed executable
 pub fn get_shell_safe_zed_path() -> anyhow::Result<String> {
-    let zed_path = std::env::current_exe()
-        .context("Failed to determine current zed executable path.")?
-        .to_string_lossy()
-        .trim_end_matches(" (deleted)") // see https://github.com/rust-lang/rust/issues/69343
-        .to_string();
-
-    // As of writing, this can only be fail if the path contains a null byte, which shouldn't be possible
-    // but shlex has annotated the error as #[non_exhaustive] so we can't make it a compile error if other
-    // errors are introduced in the future :(
-    let zed_path_escaped =
-        shlex::try_quote(&zed_path).context("Failed to shell-escape Zed executable path.")?;
-
-    Ok(zed_path_escaped.to_string())
+    let zed_path =
+        std::env::current_exe().context("Failed to determine current zed executable path.")?;
+
+    Ok(zed_path
+        .try_shell_safe()
+        .context("Failed to shell-escape Zed executable path.")?
+        .trim_end_matches(" (deleted)") // See https://github.com/rust-lang/rust/issues/69343
+        .to_string())
 }
 
 /// Returns a shell escaped path for the zed cli executable, this function
```

Comment:
Nit: is this doc comment outdated? The returned path is no longer shell-escaped.